### DISCUSSION TITLE
Triton OOMs for 16B model if asked for many samples

### DISCUSSION THREAD

**Discussion by denisyarats** [moyix/fauxpilot#90] [Q&A]
I run inference on a A100 80GB gpu with a 16B model and I noticed that I have to be careful when I ask to generate many samples. With a large enough number the FT backend OOMs. It would be better if it can just drop the request without dying.

Does anyone know if there is a way to configure this in Triton? I guess in can be implemented on the proxy level too.

**Comment by moyix** [comment]
Yeah, I have a bug report open with the Triton fastertransformer_backend asking if they can fix this, but haven't seen any response :( https://github.com/triton-inference-server/fastertransformer_backend/issues/29

The core issue is that FasterTransformer checks for out of memory using `assert` which of course just crashes the program when the assertion fails.

**Comment by denis-pplx** [comment]
Thank you for the reply!

This is crazy that they have an assert to handle this...

Do you have any recommendations on how to deal with this? Does it make sense to try to catch this in the python proxy?

**Reply by leemgs** [reply]
BTW, I am not sure if it is possible to deal with FasterTransformer's memory leak problem in Python Proxy.

**Reply by denisyarats** [reply]
Can you elaborate why? Obviously it won't be perfect, but you can imagine checking the batch size in python and if it is too large just drop the request?

**Reply by moyix** [reply]
It should be possible to check if the batch will fit in memory, yes. I've just never sat down and tried to work out exactly what the calculation is. Of course, if you have multiple requests going to the server simultaneously then all bets are off.

One thing we can do as a stopgap measure is set the Triton+FT container to auto-restart in `docker_compose.yml`, and have the proxy detect the failure and return an error 500 or something like that.

**Comment by leemgs** [comment]
When I ran the Fox Pilot Server on a 16B model with Nvidia GPU A100 (80GB of VRAM), I needed at least 32GB of VRAM. FYI

**Comment by marcoaleixo** [comment]
sorry for raising this old one but any news on this @denisyarats ?
